pkg/object/ingresscontroller/k8s.go

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright (c) 2017, MegaEase
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ingresscontroller
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/megaease/easegress/pkg/logger"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/informers"
+	corev1 "k8s.io/client-go/informers/core/v1"
+	"k8s.io/client-go/informers/internalinterfaces"
+	networkingv1 "k8s.io/client-go/informers/networking/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+
+	apicorev1 "k8s.io/api/core/v1"
+	apinetv1 "k8s.io/api/networking/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+const (
+	resyncPeriod   = 10 * time.Minute
+	defaultTimeout = 5 * time.Second
+)
+
+func isResourceEqual(oldObj, newObj interface{}) bool {
+	if oldObj == nil || newObj == nil {
+		return true
+	}
+
+	if oldObj.(metav1.Object).GetResourceVersion() == newObj.(metav1.Object).GetResourceVersion() {
+		return false
+	}
+
+	if ep, ok := oldObj.(*apicorev1.Endpoints); ok {
+		return isEndpointsEqual(ep, newObj.(*apicorev1.Endpoints))
+	}
+
+	return true
+}
+
+func isEndpointsEqual(a, b *apicorev1.Endpoints) bool {
```

Comment:
According to this function name `isEndpointsEqual`, the logic should be
`if equal returns true, else return false`, right? But it looks the opposite here.